### PR TITLE
Reference for `conan.tools.microsoft` classes and functions

### PR DESCRIPTION
**Pull request by franramirez688**
Related to https://github.com/conan-io/conan/pull/11283

### CODE REVIEW

File: reference/tools/microsoft/helpers.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+check_min_vs
+---------------
+
+
+.. currentmodule:: conan.tools.microsoft.visual
+
+.. autofunction:: check_min_vs
+
+
+vcvars_command
```
Comment (memsharded) [inline reference/tools/microsoft/helpers.rst:15]: Not for being public yet, I think

File: reference/tools/microsoft/msbuild.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+========
+
+The ``MSBuild`` build helper is a wrapper around the command line invocation of MSBuild. It abstracts the
+calls like ``msbuild "MyProject.sln" /p:Configuration=<conf> /p:Platform=<platform>`` into Python method ones.
+
+This helper can be used like:
+
+.. code:: python
+
+    from conans import conanfile
```
Comment (memsharded) [inline reference/tools/microsoft/msbuild.rst:14]: Wrong import!
```suggestion
    from conan import ConanFile
```

File: reference/tools/microsoft/msbuilddeps.rst [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    [generators]
+    MSBuildDeps
+
+And it can also be fully instantiated in the conanfile ``generate()`` method:
+
+.. code-block:: python
+    :caption: conanfile.py
+
+    from conans import ConanFile
```
Comment (memsharded) [inline reference/tools/microsoft/msbuilddeps.rst:32]:
```suggestion
    from conan import ConanFile
```

File: reference/tools/microsoft/msbuilddeps.rst [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+If your Visual Studio project defines custom configurations, like ``ReleaseShared``, or ``MyCustomConfig``,
+it is possible to define it into the ``MSBuildDeps`` generator, so different project configurations can
+use different set of dependencies. Let's say that our current project can be built as a shared library,
+with the custom configuration ``ReleaseShared``, and the package also controls this with the ``shared``
+option:
+
+.. code-block:: python
+
+    from conans import ConanFile
```
Comment (memsharded) [inline reference/tools/microsoft/msbuilddeps.rst:84]:
```suggestion
    from conan import ConanFile
```

File: reference/tools/microsoft/msbuildtoolchain.rst [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+    [generators]
+    MSBuildToolchain
+
+
+And it can also be fully instantiated in the conanfile ``generate()`` method:
+
+.. code:: python
+
+    from conans import ConanFile
```
Comment (memsharded) [inline reference/tools/microsoft/msbuildtoolchain.rst:30]:
```suggestion
    from conan import ConanFile
```

File: reference/tools/microsoft/helpers.rst (hunk 1 above)
Comment (franramirez688) [reply reference/tools/microsoft/helpers.rst:15]: I recall a couple of recipes using it in conan-center-index...

Comment (memsharded) [reply reference/tools/microsoft/helpers.rst:15]: Yes, but as you can see, this one doesn't have the ``conanfile`` as first argument. That is a hint that this is not ready to be made public (as it won't be able to use ``conf`` if needed). Lets hide it, so these use cases arises while the migration and we can check them.

Comment (franramirez688) [reply reference/tools/microsoft/helpers.rst:15]: oh, ok. That's right

File: reference/tools/microsoft/msbuild.rst [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+========
+
+The ``MSBuild`` build helper is a wrapper around the command line invocation of MSBuild. It abstracts the
+calls like ``msbuild "MyProject.sln" /p:Configuration=<conf> /p:Platform=<platform>`` into Python method ones.
+
+This helper can be used like:
+
+.. code:: python
+
+    from conan import conanfile
```
Comment (memsharded) [inline reference/tools/microsoft/msbuild.rst:14]:
```suggestion
    from conan import ConanFile
```

Still wrong! :)

File: reference/tools/microsoft/msbuildtoolchain.rst [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            tc = MSBuildToolchain(self)
+            tc.generate()
+
+
+The ``MSBuildToolchain`` will generate three files after a ``conan install`` command:
+
+.. code-block:: bash
+
+    $ conan install conanfile.py # default is Release
+    $ conan install conanfile.py -s build_type=Debug
```
Comment (memsharded) [inline reference/tools/microsoft/msbuildtoolchain.rst:46]:
```suggestion
    $ conan install . # default is Release
    $ conan install . -s build_type=Debug
```

File: reference/tools/microsoft/msbuilddeps.rst [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+            ms.generate()
+
+
+The ``MSBuildDeps`` generator is a multi-configuration generator, and generates different files for any different
+Debug/Release configuration. For instance, running these commands:
+
+.. code-block:: bash
+
+    $ conan install conanfile.py  # default is Release
+    $ conan install conanfile.py -s build_type=Debug
```
Comment (memsharded) [inline reference/tools/microsoft/msbuilddeps.rst:50]:
```suggestion

    $ conan install .  # default is Release
    $ conan install . -s build_type=Debug
```

**Review by memsharded — APPROVED** [review]

**Pull request merged**